pkg/globalnet/controllers/uninstall.go

```diff
@@ -216,3 +218,32 @@ func RemoveGlobalIPAnnotationOnNode(cfg *rest.Config) {
 
 	logger.Infof("Successfully removed globalIP annotation from node %q", nodeName)
 }
+
+func RemoveStaleInternalServices(config *syncer.ResourceSyncerConfig) error {
+	_, gvr, err := util.ToUnstructuredResource(&corev1.Service{}, config.RestMapper)
+	if err != nil {
+		return errors.Wrap(err, "error converting resource")
+	}
+
+	services := config.SourceClient.Resource(*gvr)
+
+	svcList, err := services.Namespace(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
```

Comment:
Since we're only interested in globalnet internal services, we can optimize the `List` call by using a label selector for the `InternalServiceLabel`:

```suggestion
	svcList, err := services.Namespace(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
		LabelSelector: InternalServiceLabel,
	})
```

Reply:
This is a good optimization.